### PR TITLE
Delivery/availability for runs

### PR DESCRIPTION
**Pull request by mbertrand**
### What are the relevant tickets?
Backend part of https://github.com/mitodl/hq/issues/5307

### Description (What does it do?)
-Run `docker compose up` or `docker compose run --rm web ./manage.py migrate`.  Migrations should succeed without errors.
- Check that any OCW courses in your database have runs with a `delivery` value equal to their parent course (usually `["online", "offline"]`.  Ditto non-ocw courses (usually `["online"]`).
- Check that any course/program resources in your database have runs with a `availability` value equal to their parent course.
- Run the following , they should all still work (You'll need to populate all the relevant env settings first):
  ```
  ./manage.py backpopulate_micromasters_data
  ./manage.py backpopulate_mitxonline_data
  ./manage.py backpopulate_mit_edx_data
  ./manage.py backpopulate_sloan_data
  ./manage.py backpopulate_xpro_data
  ./manage.py backpopulate_prolearn_data
  ./manage.py backpopulate_oll_data
  ./manage.py backpopulate_ocw_data --course-name 21w-758-genre-fiction-workshop-fantasy-fall-2016 --skip-contentfiles
  ```
Afterward, search for http://open.odl.local:8062/search?q=%22Accelerating+Digital+Transformation+with+Algorithmic+Business+Thinking%22 - open the drawer for that course and inspect the api result, the "delivery"  attributes should have some differing values by run.

### CODE REVIEW

**Review by rhysyngsun — CHANGES_REQUESTED** [review]
One small piece of feedback but otherwise is looking good.

File: learning_resources/etl/loaders.py [hunk 1]
```diff
@@ -130,13 +139,14 @@ def load_next_start_date_and_prices(
...
     )
+    resource.availability = best_run.availability if best_run else resource.availability
     resource.prices = (
         best_run.prices
         if resource.certification and best_run and best_run.prices
         else []
     )
     resource.save()
-    return resource.next_start_date, resource.prices
+    return resource.next_start_date, resource.prices, resource.availability
```
[the remark this answers is not in the document]
Comment (mbertrand) [reply learning_resources/etl/loaders.py]: Thanks, done.

File: learning_resources/etl/constants.py [hunk 2]
```diff
@@ -106,3 +109,10 @@ class ContentTagCategory(ExtendedEnum):
     "Assignments": ContentTagCategory.problem_sets.value,
     # Can add more here if ever needed, in format tag_name:category
 }
+
+
+@dataclass
+class ResourceNextRunConfig:
+    next_start_date: datetime = None
+    prices: list[Decimal] = list
```
Comment (rhysyngsun) [inline learning_resources/etl/constants.py:117]: Should use `dataclasses.field` to specify a `[]` default, this is currently assigning the function `list` as the default:
```suggestion
    prices: list[Decimal] = field(default_factory=list)
```

**Review by rhysyngsun — CHANGES_REQUESTED** [review]

File: learning_resources/etl/constants.py (hunk 2 above)
Comment (mbertrand) [reply learning_resources/etl/constants.py]: done

**Review by rhysyngsun — APPROVED** [review]
LGTM

**Pull request merged**